plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/XBuildManager.java

```diff
@@ -271,7 +274,7 @@ public XBuildable getIncrementalGenerateBuildable(List<URI> dirtyFiles, List<URI
 			throw ce;
 
 		} catch (Exception e) {
-			// recover
+			// recover and also discard the build queue - state is undefined afterwards.
 			this.dirtyFiles.clear();
 			this.deletedFiles.clear();
```

Comment:
How do we get back into a well-defined state after this?

Reply:
Some months ago I modified code here but we need to create some tests for this scenario (although this would be difficult).